gr-fec/lib/cc_encoder_impl.h

```diff
@@ -34,7 +34,7 @@ class FEC_API cc_encoder_impl : public cc_encoder
     unsigned int d_rate;
     unsigned int d_k;
     std::vector<int> d_polys;
-    int d_start_state;
+    unsigned int d_start_state;
```

Comment:
Note that ``start_state`` is passed as ``signed int`` (``signed`` added for clarity :wink:). For consistency that should be changed as well – but this is an API change. I haven’t changed the type of ``d_start_state`` in #3103 as I wanted to backport the fixes therein to maint-3.[78].

There are also other parameters whose type should be changed to ``unsigned`` .

Reply:
Are you proposing that I make those API changes in this pull request?

Reply:
Yeah, so hard call to make: on one hand, let's not change API arbitrarily, but really, the constructor has no business taking a signed int for something that's a bitfield.

So, considering that, for `master`, the saner thing might actually be to change the API to take an `unsigned int`. 

The fact that the variable type was changed to `signed` (instead of `signed int`) slipped my attention; that'd be the only place all over GNU Radio where we do that. I'm not changing that now, but it doesn't really strike me as aesthetically fitting ;)

Reply:
GR has -so- many API changes needed to go from `singed` to `unsigned` ... I'd prefer to not do that here but rather as a whole shebang change. Since we have a check for the max value of `d_start_state` already, I don't see a need to do more than what's in this PR right now. It's a simple and needed fix as is IMHO.